src/pybind/mgr/dashboard_v2/frontend/src/app/shared/components/table/table.component.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  Component, EventEmitter, OnInit, Input, Output, ViewChild, OnChanges, ComponentFactoryResolver, Type
+} from '@angular/core';
+import {DatatableComponent, TableColumn} from '@swimlane/ngx-datatable';
+import {TableDetailsDirective} from './table-details.directive';
+
+@Component({
+  selector: 'cd-table',
+  templateUrl: './table.component.html',
+  styleUrls: ['./table.component.scss']
+})
+export class TableComponent implements OnInit, OnChanges {
+  @ViewChild(DatatableComponent) table: DatatableComponent;
+  @ViewChild(TableDetailsDirective) detailTemplate: TableDetailsDirective;
+
+  @Input() data: any[]; // This is the array with the items to be shown
+  @Input() columns: TableColumn[]; // each item -> { prop: 'attribute name', name: 'display name' }
+  @Input() detailsComponent?: string; // name of the component fe 'TableDetailsComponent'
+  @Input() header? = true;
+
+  @Output() fetchData = new EventEmitter(); // Should be the function that will update the input data
```

Comment:
Wouldn't it make sense to declare it as ``@Output() fetchData = new EventEmitter<string>();``. This way it is clear that a string is submitted as function argument by the ``emit`` function.

Reply:
Changed the usage of "fetchData" to not call  it with a string